### PR TITLE
Move EIP-712 to Last Call

### PR DESCRIPTION
**Pull request by frangio**
There have been no substantial changes to this EIP in over a year.

This PR also includes non-normative changes to adapt the EIP to new formatting requirements. A mention of "EIP-719" was removed because said EIP doesn't exist. The "Implementation" section was removed as it consisted of an implementation log that doesn't belong in the EIP format.

### CODE REVIEW

**Comment by xinbenlv** [comment]
So looking forward to it

**Comment by MicahZoltu** [comment]
Please see the files tab where a new bot has made a bunch of comments (they don't currently trigger any email/notification).

**Comment by frangio** [comment]
@MicahZoltu If I understand correctly these errors are due to EIP format requirements that have changed or become stricter since this document was written.

Should I make the changes required to fit the new format? Will those interfere with the attempt to move to Last Call status?

**Comment by SamWilsn** [comment]
> Should I make the changes required to fit the new format?

Yes please!

> Will those interfere with the attempt to move to Last Call status?

Nope. Status changes are usually where we make sure EIPs are up to date with the latest format requirements.

Do note, however, that substantial changes _after_ the EIP enters last call might require resetting the last call deadline.

File: EIPS/eip-712.md
Comment (frangio) [inline EIPS/eip-712.md:37]: It's asking for a link here but it's a self-referential link which I don't think makes sense.

Comment (SamWilsn) [reply EIPS/eip-712.md:37]: Hm, I'll open an issue in the bot to decide how to handle self-references.

For now, maybe:

```suggestion
Here we outline a scheme to encode data along with its structure which allows it to be displayed to the user for verification when signing. Below is an example of what a user could be shown when signing a message according to this proposal.
```

File: EIPS/eip-712.md
Comment (frangio) [inline EIPS/eip-712.md:373]: (Referring to the line below this one. Couldn't comment directly on it.)

There is a reference and link to "[EIP-719](https://github.com/ethereum/EIPs/issues/719
)" which doesn't really exist. What would you suggest? @MicahZoltu Any suggestions being the author of that issue?

Comment (SamWilsn) [reply EIPS/eip-712.md:376]: Hm, while I'd recommend waiting for Micah's input, I think you can probably just remove this line. Since it isn't mentioned in the Specification section as a permitted field, this seems like a contradiction anyway.

File: EIPS/eip-712.md
Comment (SamWilsn) [inline EIPS/eip-712.md:37]:
```suggestion
![eth_sign screenshot](../assets/eip-712/eth_sign.png)
```

File: EIPS/eip-712.md
Comment (xinbenlv) [reply EIPS/eip-712.md:36]: Yes I am commenting on the latest snapshot to my knowledge.

I know that you've already converted the original 2 lines below "Simple Summary" section to merge with "Abstract". I was proposing the first 2 lines in the new Abstract to be moved to "Motivation"

File: EIPS/eip-712.md
Comment (xinbenlv) [reply EIPS/eip-712.md:3]: Thank you for being open to it.

Since, IIUC the goal is to have a standardized way of hashing and signing, to avoid everyone re-inventing the wheels, I think it's more crypto-specific than crypto-generalized.

Option 1. *Ethereum typed structured data ~~hashing and~~ signing* (signing alone is likely to imply there will be some form of hashing)
Option 2. `Ethereum typed ~~structured~~ data hashing and signing` (typed usually imply structured)
Option 3. `Ethereum ~~typed~~ structured data hashing and signing` (structured usually imply typed)

Comment (SamWilsn) [reply EIPS/eip-712.md:3]: All EIPs are Ethereum Improvement Proposals. Ethereum is redundant in the title.

File: EIPS/eip-712.md
Comment (MicahZoltu) [reply EIPS/eip-712.md:380]: There is no EIP-719, so you can't/shouldn't link to it.  I would *love it* if someone would create an EIP-719 and champion it and push it through to final, but at the moment that isn't happening so removing the line seems like the best course of action.

File: EIPS/eip-712.md
Comment (frangio) [reply EIPS/eip-712.md:36]: Got it, it's a good idea.

**Comment by frangio** [comment]
This should be ready to merge once one of the authors approves it.

@LogvinovLeon @recmo @dekz

**Review by LogvinovLeon — APPROVED** [review]

**Comment by frangio** [comment]
@SamWilsn This has got author approval. Just need editor approval now. :slightly_smiling_face:

File: EIPS/eip-712.md
Comment (SamWilsn) [inline EIPS/eip-712.md:393]: Eep! Just realized this EIP is missing the `Security Considerations` section. It's always mandatory, but for an EIP about signing data, I think it's _especially_ mandatory.

It'd be good to note down concerns about signature reuse/nonces at the very least.

Comment (frangio) [reply EIPS/eip-712.md:393]: Ok. That's disappointing! I will see what I can whip up.

Comment (SamWilsn) [reply EIPS/eip-712.md:393]: Thanks for taking the time to push this EIP through. I know it can be a gruelling process.

Comment (MicahZoltu) [reply EIPS/eip-712.md:393]: @SamWilsn Any idea why EIPW didn't flag this?  I thought this was one of the first checks you implemented?

Comment (SamWilsn) [reply EIPS/eip-712.md:393]: I implemented section order, and no unknown sections. Completely forgot required sections. It's implemented and deployed now.

Still need to do duplicate sections.

**Comment by frangio** [comment]
Ok I've moved the existing section about replay attacks into a Security Considerations section, I think we should be good now.

cc @SamWilsn

File: EIPS/eip-712.md
Comment (xinbenlv) [inline EIPS/eip-712.md:394]: Since you mentioned the replay attack, it's probably also worthy of mentioning that in many case once a message is signed, they are no guarantee their distribution will be restricted or they delivery will be ensured and confirmed, therefore the applications need to consider such cases to ensure safety.

Comment (frangio) [reply EIPS/eip-712.md:394]: What do you mean by "restricted distribution"?

Comment (Pandapip1) [reply EIPS/eip-712.md:394]: > What do you mean by "restricted distribution"?

I imagine he means that the signature should be assumed to possibly become public knowledge.

Comment (xinbenlv) [reply EIPS/eip-712.md:394]: @Pandapip1 is right. Yes, the signer shall expect the signature MAY AND MAY NOT be public knowledge.

**Comment by frangio** [comment]
I've added a small note concerning the distribution of the signature and in particular frontrunning attacks as suggested by @xinbenlv, which should be uncontroversial.

But this PR really shouldn't result in any further modifications as it already has author approval. Please merge as soon as possible!

**Review by Pandapip1 — APPROVED** [review]

**Review by Pandapip1 — APPROVED** [review]
Let's try this again.

**Comment by Pandapip1** [comment]
@frangio please push an empty commit.

**Review by Pandapip1 — APPROVED** [review]
CI should be fixed

**Pull request merged**
